File: Strings/Easy/romantointeger/romantoint.cpp
#include <iostream>
#include <unordered_map>

int romanToInt(const std::string &s)
{
    // Define a unordered_map mapping Roman numerals to their integer values
    std::unordered_map<char, int> romanMap{
        {'I', 1},
        {'V', 5},
        {'X', 10},
        {'L', 50},
        {'C', 100},
        {'D', 500},
        {'M', 1000}};

    int result = 0; // Initialize the result variable to store the total integer value

    // Loop through the Roman numeral string from left to right
    for (int i = 0; i < s.length(); ++i)
    {
        // If the current numeral is smaller than the next numeral, subtract its value
        if (i < s.length() - 1 && romanMap[s[i]] < romanMap[s[i + 1]])
        {
            result -= romanMap[s[i]];
        }
        else
        {
            result += romanMap[s[i]];
        }
    }

    return result;
}

int main()
{
    // Test the function
    std::string romanNumeral;
    std::cout << "Enter a Roman numeral: ";
    std::cin >> romanNumeral;

    int integerValue = romanToInt(romanNumeral);
    std::cout << "The integer value of " << romanNumeral << " is: " << integerValue << std::endl;

    return 0;
}

File: Strings/Easy/romantointeger/romantoint.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class romantoint {

    public static int romanToInt(String s) {
        // Define a map mapping Roman numerals to their integer values
        Map<Character, Integer> romanMap = new HashMap<>();
        romanMap.put('I', 1);
        romanMap.put('V', 5);
        romanMap.put('X', 10);
        romanMap.put('L', 50);
        romanMap.put('C', 100);
        romanMap.put('D', 500);
        romanMap.put('M', 1000);

        int result = 0; // Initialize the result variable to store the total integer value

        // Loop through the Roman numeral string from left to right
        for (int i = 0; i < s.length(); i++) {
            // If the current numeral is smaller than the next numeral, subtract its value
            if (i < s.length() - 1 && romanMap.get(s.charAt(i)) < romanMap.get(s.charAt(i + 1))) {
                result -= romanMap.get(s.charAt(i));
            } else {
                result += romanMap.get(s.charAt(i));
            }
        }

        return result;
    }

    public static void main(String[] args) {
        try (// Test the function
                Scanner scanner = new Scanner(System.in)) {
            System.out.print("Enter a Roman numeral: ");
            String romanNumeral = scanner.nextLine();

            int integerValue = romanToInt(romanNumeral.toUpperCase());
            System.out.println("The integer value of " + romanNumeral + " is: " + integerValue);
        }
    }
}
